apps/etterlatte-saksbehandling-ui/client/src/components/oppgavebenk/gosys/FerdigstillGosysOppgave.tsx

```diff
@@ -0,0 +1,52 @@
+import { ferdigstilleGosysOppgave, OppgaveDTO } from '~shared/api/oppgaver'
+import { useApiCall } from '~shared/hooks/useApiCall'
+import { isFailure, isPending, isSuccess } from '~shared/api/apiUtils'
+import { Alert, Button, Loader } from '@navikt/ds-react'
+import { FlexRow } from '~shared/styled'
+import { GosysActionToggle } from '~components/oppgavebenk/oppgaveModal/GosysOppgaveModal'
+
+export const FerdigstillGosysOppgave = ({
+  oppgave,
+  setToggle,
+}: {
+  oppgave: OppgaveDTO
+  setToggle: (toggle: GosysActionToggle) => void
+}) => {
+  const [ferdigstillResult, ferdigstillOppgave] = useApiCall(ferdigstilleGosysOppgave)
+
+  const ferdigstill = () =>
+    ferdigstillOppgave({ oppgaveId: oppgave.id, versjon: oppgave.versjon || 0 }, () => {
+      setTimeout(() => window.location.reload(), 2000)
+    })
+
+  if (isSuccess(ferdigstillResult)) {
+    return (
+      <Alert variant="success">
+        Oppgaven ble ferdigstilt. Henter oppgaver på nytt <Loader />
+      </Alert>
+    )
+  } else if (isFailure(ferdigstillResult)) {
+    return (
+      <Alert variant="error">
+        {ferdigstillResult.error.detail || 'Ukjent feil oppsto ved ferdigstilling av oppgave'}
+      </Alert>
+    )
+  }
```

Comment:
Kanskje dette kan flyttes i et eget komponent som tar inn `ferdigstillResult` som prop, for å så flytte den inn i return statementen. Så blir det litt bedre skille mellom ting 😃

Reply:
ryddet litt 👍